pkg/obfuscator/tracker.go

```diff
@@ -15,11 +15,11 @@ type ReplacementTracker interface {
 	Initialize(replacements map[string]string)
 
 	// Report returns a mapping of strings which were replaced.
-	Report() map[string]string
+	Report() map[string]Replacement
 
 	// AddReplacement will add a replacement along with its original string to the report.
 	// If there is an existing value that does not match the given replacement, it will exit with a non-zero status.
-	AddReplacement(original string, replacement string)
+	AddReplacement(original string, replacement Replacement)
```

Comment:
leave that at (string, string), if you want to convey a count with multiple replacements then we can add another method that also includes a count, so (string, string, int).

I think it might not be easy for things like the keyword replacer that just replace in bulk - let's see